using-timescaledb/actions.md

```diff
@@ -0,0 +1,255 @@
+# User-defined actions
+
+User-defined actions allow you to run functions and procedures implemented in a
+language of your choosing on a schedule within TimescaleDB. This allows
+automatic periodic tasks that are not covered by existing policies and
+even enhancing existing policies with additional functionality.
+
+### Creating a procedure for an action [](create-procedure)
+
+The signature for actions is (job_id INT, config JSONB). It can either
+be implemented as [function](postgres-createfunction) or
+[procedure](postgres-createprocedure).
+The content of the config JSONB is completely up to the job and may
+also be NULL if no variables are required.
+
+Template for a procedure.
+
+```sql
+CREATE OR REPLACE PROCEDURE custom_job(job_id int, config jsonb) LANGUAGE PLPGSQL AS
+$$
+BEGIN
+	RAISE NOTICE 'Executing job % with config %', job_id, config;
+END
+$$;
+```
+
+### Registering your action [](register)
+
+In order to register your action for execution within TimescaleDB's
+job scheduler, you next need to `add_job` with the name of your action,
+as well as the schedule on which it is run.
+
+When registered, the action's `job_id` and `config` are stored in the
+TimescaleDB catalog. The `config` JSONB can be modified with `alter_job`.
+`job_id` and `config` will be passed as arguments when the procedure is
+executed as background process or when expressly called with `run_job`.
+
+```sql
+SELECT add_job('custom_job','1h');
+```
+
+Register the created job with the automation framework. `add_job` returns the job_id
+which can be used to execute the job manually with `run_job`.
```

Comment:
The description of an example code is normally preceding the code. Putting afterwards is inconsistent with the rest of the documentation.